index.ts

```diff
@@ -822,6 +876,51 @@ class IntegrationAPI extends EventEmitter {
   public updateEntityAttributes(entityId: string, attributes: { [key: string]: string | number | boolean }): boolean {
     return this.#configuredEntities.updateEntityAttributes(entityId, attributes);
   }
+
+  public async generateOauth2AuthUrl(): Promise<{ auth_url: string }> {
+    const conn = this.#getCoreClient();
+    if (!conn) {
+      log.warn("generateOauth2AuthUrl: expected 1 client connected");
+      return Promise.reject("expected 1 client connected");
+    }
+    return this.#sendRequest(conn, generateRandomId(), api.MsgEvents.GenerateOauth2AuthUrl);
+  }
+
+  public async createOauth2Config(data: { [key: string]: any }) {
+    const conn = this.#getCoreClient();
+    if (!conn) {
+      log.warn("createOauth2Config: expected 1 client connected");
+      return Promise.reject("expected 1 client connected");
+    }
+
+    return this.#sendRequest(conn, generateRandomId(), api.MsgEvents.CreateOauth2Cfg, data);
+  }
+
+  public async deleteOauth2Token(data: { [key: string]: any }) {
+    const conn = this.#getCoreClient();
+    if (!conn) {
+      log.warn("deleteOauth2Token: expected 1 client connected");
+      return Promise.reject("expected 1 client connected");
+    }
+    return this.#sendRequest(conn, generateRandomId(), api.MsgEvents.DeleteOauth2Token, data);
+  }
+
+  public async getOauth2Token(data: { token_id: string; force_refresh: boolean }) {
+    const conn = this.#getCoreClient();
+    if (!conn) {
+      log.warn("getOauth2Token: expected 1 client connected");
+      return Promise.reject("expected 1 client connected");
+    }
+    return this.#sendRequest(conn, generateRandomId(), api.MsgEvents.GetOauth2Token, data);
+  }
+
+  #getCoreClient() {
+    if (this.#clients.size !== 1) {
+      log.warn("getCoreClient: more than one client connected, total %s", this.#clients.size);
+      return null;
+    }
+    return [...this.#clients.keys()][0];
```

Comment:
I'm not a TS expert, but isn't that a potential runtime error if there are no clients connected?